postgres/tests/test_statements.py

```diff
@@ -497,6 +500,220 @@ def test_statement_samples_collect(
+    check._connect()
+
+    conn = psycopg2.connect(host=HOST, dbname=dbname, user=user, password=password)
+
+    # we are able to see the full query (including the raw parameters) in pg_stat_activity because psycopg2 uses
+    # the simple query protocol, sending the whole query as a plain string to postgres.
+    # if a client is using the extended query protocol with prepare then the query would appear as
+    # leave connection open until after the check has run to ensure we're able to see the query in
+    # pg_stat_activity
+    try:
+        # turn off auto commit so we can manage transactions manually
+        # this should allow us to catch queries in the 'idle in transaction' state
+        conn.autocommit = False
+        conn.cursor().execute(query, (arg,))
+        check.check(dbm_instance)
+        dbm_activity_event = aggregator.get_event_platform_events("dbm-activity")
+
+        if POSTGRES_VERSION.split('.')[0] == "9" and pg_stat_activity_view == "pg_stat_activity":
+            # cannot catch any queries from other users
+            # only can see own queries
+            return
+
+        event = dbm_activity_event[0]
+        assert event['host'] == "stubbed.hostname"
+        assert event['ddsource'] == "postgres"
+        assert event['dbm_type'] == "activity"
+        assert event['ddagentversion'] == datadog_agent.get_version()
+        assert len(event['postgres_activity']) > 0
+        # find bob's query.
+        bobs_query = None
+        for query_json in event['postgres_activity']:
+            if 'usename' in query_json and query_json['usename'] == "bob":
+                bobs_query = query_json
+        assert bobs_query is not None
+
+        for key in expected_out:
+            assert expected_out[key] == bobs_query[key]
+        for val in expected_keys:
+            assert val in bobs_query
+
+        assert 'query' not in bobs_query
+
+        expected_tags = dbm_instance['tags'] + [
+            'port:{}'.format(PORT),
+        ]
+
+        # check postgres_connections are set
+        assert len(event['postgres_connections']) > 0
+        # find bob's connections.
+        bobs_conns = None
+        for query_json in event['postgres_connections']:
+            if 'usename' in query_json and query_json['usename'] == "bob":
+                bobs_conns = query_json
+        assert bobs_conns is not None
+
+        for key in expected_conn_out:
+            assert expected_conn_out[key] == bobs_conns[key]
+
+        assert event['ddtags'] == expected_tags
+
+    finally:
+        conn.close()
+
+
+@pytest.mark.parametrize(
+    "active_rows",
+    [
+        (
+            [
+                {
+                    'datname': 'datadog_test',
+                    'usename': 'newbob',
+                    'xact_start': datetime.datetime(2021, 9, 23, 23, 21, 21, 669330, tzinfo=DEFAULT_TZ_INFO),
+                    'query_start': datetime.datetime(2021, 9, 23, 23, 21, 21, 669330, tzinfo=DEFAULT_TZ_INFO),
```

Comment:
Could be helpful to hide the construction of the time away to make this easier to parse visually and immediately know which is older:

```
                    'xact_start': test_time(1),
                    'query_start': test_time(2),
... 
```

Reply:
I attempted this here https://github.com/DataDog/integrations-core/pull/9866/commits/1e390b8f9c6beb7f553992dee41d4e1891b123d8

I couldn't get this to work by calling a single function with an input, but this approach seems fine?